src/main/java/com/scottlogic/DatePostSorter.java

```diff
@@ -0,0 +1,25 @@
+package com.scottlogic;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class DatePostSorter implements PostSorter{
+
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder sortOrder) {
+        if (inputList == null) {
+            return null;
+        } else if (inputList.isEmpty()) {
+            return inputList;
+        }
+        List<UserPost> userPosts = new ArrayList<UserPost>(inputList) ;
```

Comment:
question, why are you creating a copy of the userPosts before you sort? Can you not sort the inputList directly?

Reply:
That's what I was doing! I was sorting directly the inputList. But I got told in the previous pull request that Jamie reviewed that it was not secure to sort the inputList. Maybe I misunderstood